### PR TITLE
refactor: [M3-7576] - DebouncedSearchTextfield and EditableText v7 storybook migrations

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
- Storybook migrations for Debounced search textfield & editable text
- added tests as well

### Verification steps 
- yarn test DebouncedSearchTextField
- yarn test EditableText
- http://localhost:6006/?path=/docs/components-editable-text--documentation
- http://localhost:6006/?path=/docs/components-search--documentation

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/DebouncedSearchTextField/StoryComponents/TextFieldExample.tsx [hunk 1]
```diff
@@ -1,69 +0,0 @@
-import { action } from '@storybook/addon-actions';
```
Comment (coliu-akamai) [inline packages/manager/src/components/DebouncedSearchTextField/StoryComponents/TextFieldExample.tsx:1]: moved this to be a part of `DebouncedSearchTextfield.stories.tsx`

File: packages/manager/vite.config.ts.timestamp-1703111137009-e62fdfbfec8b3.mjs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// vite.config.ts
```
Comment (coliu-akamai) [inline packages/manager/vite.config.ts.timestamp-1703111137009-e62fdfbfec8b3.mjs:1]: oh no! lemme try and remove this??

File: packages/manager/src/components/DebouncedSearchTextField/DebouncedSearchTextField.tsx [hunk 3]
```diff
@@ -7,12 +7,21 @@ import { InputAdornment } from 'src/components/InputAdornment';
 import { TextField, TextFieldProps } from 'src/components/TextField';
 import { usePrevious } from 'src/hooks/usePrevious';
 
-interface DebouncedSearchProps extends TextFieldProps {
+export interface DebouncedSearchProps extends TextFieldProps {
   className?: string;
+  /**
+   * Interaval of time that passes before search queries are accepted. Default is 400
+   */
```
Comment (bnussman-akamai) [inline packages/manager/src/components/DebouncedSearchTextField/DebouncedSearchTextField.tsx:14]: Spelling of `Interval ` and maybe the `@default` makes sense here?

```suggestion
  /**
   * Interval in milliseconds of time that passes before search queries are accepted.
   * @default 400
   */
```

**Review by bnussman-akamai — APPROVED** [review]
Components worked as expected ✅

**Review by mjac0bs — APPROVED** [review]
Stories render and look good in Storybook; both unit tests pass. I pushed one commit to clean up the readability of the stories a little. 🚀  I'm going to merge this once CI finishes.

**Pull request merged**